File: CS207b/L/w02/src_java/SortedList.java
package CS207b.L.w02.src_java;

import java.util.ArrayList;

public class SortedList {
    private ArrayList<Integer> list;

    public SortedList(ArrayList<Integer> list) {
        this.list = new ArrayList<Integer>(list);
    }

    public ArrayList<Integer> getList() {
        return list;
    }

    public void selectionSort() {
        int n = list.size();
        for (int i = 0; i < n - 1; i++) {
            int min_idx = i;
            for (int j = i + 1; j < n; j++)
                if (list.get(j) < list.get(min_idx))
                    min_idx = j;
            int temp = list.get(min_idx);
            list.set(min_idx, list.get(i));
            list.set(i, temp);
        }
    }

    public void bubbleSort() {
        int n = list.size();
        for (int i = 0; i < n - 1; i++)
            for (int j = 0; j < n - 1 - i; j++)
                if (list.get(j) > list.get(j + 1)) {
                    int temp = list.get(j);
                    list.set(j, list.get(j + 1));
                    list.set(j + 1, temp);
                }
    }

    public void countingSort(int exp) {
        System.out.println("Counting Sort");
    }

    public void radixSort() {
        System.out.println("Radix Sort");
    }

    public void interchangeSort() {
        System.out.println("Interchange Sort");
    }
}

File: CS207b/L/w02/src/SortedList.py
class SortedList:
    __list = []

    def __init__(self, numbers):
        self.__list = numbers

    def selectionSort(self):
        for i in range(len(self.__list)):
            minPosition = i
            for j in range(i, len(self.__list)):
                if self.__list[j] < self.__list[minPosition]:
                    minPosition = j
            self.__list[i], self.__list[minPosition] = (
                self.__list[minPosition],
                self.__list[i],
            )
        return self.__list

    def bubbleSort(self):
        for i in range(len(self.__list)):
            for j in range(len(self.__list) - 1):
                if self.__list[j] > self.__list[j + 1]:
                    self.__list[j], self.__list[j + 1] = (
                        self.__list[j + 1],
                        self.__list[j],
                    )
        return self.__list

    def countingSort(self):
        print("Counting Sort")

    def radixSort(self):
        print("Radix Sort")

    def interchangeSort(self):
        print("Interchange Sort")

    def print(self):
        print(self.__list)
